Nos/Controller/NoteEditorController.swift

```diff
@@ -0,0 +1,252 @@
+    }
+    
+    /// Inserts a string at the given range and adds the link attribute with the given `link`.
+    private func insert(text: String, link: String, at range: NSRange) {
+        guard let textView else {
+            return
+        }
+        
+        let linkAttributes = defaultStringAttributes.merging(
+            [NSAttributedString.Key.link: link], 
+            uniquingKeysWith: { _, key in key }
+        )
+        let link = NSAttributedString(
+            string: text,
+            attributes: linkAttributes
+        )
+        
+        let attributedString = NSMutableAttributedString(attributedString: textView.attributedText)
+        attributedString.replaceCharacters(in: range, with: link)
+        textView.attributedText = attributedString
+        textView.selectedRange.location = range.location + link.length
+        isEmpty = false
+    }
+    
+    /// Takes the same arguments as `textView(_:shouldChangeTextIn:replacementText:)` and detects the case where the 
+    /// user is typing inside a link (like a mention of another author). If this is the case we remove the link 
+    /// attribute from the link and insert the given `newText` and return true. Otherwise this will return `false`.
+    private func removeLinkAttributesUnderCursor(
+        selectedRange: NSRange, 
+        in textView: UITextView, 
+        newText: String
+    ) -> Bool {
+        let attributedText = textView.attributedText!
+        if attributedText.length > 0, selectedRange.location < attributedText.length || newText.isEmpty {
+            var rangeOfLink = NSRange()
+            let location = newText.isEmpty ? max(selectedRange.location, 0) : selectedRange.location
+            let link = attributedText.attribute(
+                .link, 
+                at: location, 
+                longestEffectiveRange: &rangeOfLink, 
+                in: NSRange(location: 0, length: attributedText.length)
+            )
+            
+            if link != nil {
+                let newAttributesString = NSMutableAttributedString(attributedString: attributedText)
+                newAttributesString.removeAttribute(.link, range: rangeOfLink)
+                newAttributesString.replaceCharacters(in: selectedRange, with: newText)
+                textView.attributedText = newAttributesString
+                textView.selectedRange = NSRange(
+                    location: selectedRange.location + (newText as NSString).length, 
+                    length: 0
+                )
+                return true
+            }
+        }
+        
+        return false
+    }
+        
+    /// Call this when the user has typed a '@' and it will trigger the mentions autocomplete to open if appropriate.
+    /// - Returns: `true` if it opened the mentions autocomplete.
+    func checkForMentionsAutocomplete(in textView: UITextView, at range: NSRange) -> Bool {
+        if textView.text.count == 0 {
```

Comment:
Better use isEmpty property here as it is more suitable. I'm curious why SwiftLint didn't catch it because it normally complains